app/controllers/concerns/hal_actions/actions.rb

```diff
@@ -0,0 +1,104 @@
+module HalActions::Actions
+  def index
+    respond_with index_collection, index_options
+  end
+
+  def show
+    respond_with show_resource, show_options
+  end
+
+  def create
+    create_resource.tap do |resource|
+      consume_with_content_type! resource
+      authorize resource
+      resource.save!
+      respond_with root_resource(resource), show_options
+    end
+  end
+
+  def update
+    update_resource.tap do |resource|
+      consume_with_content_type! resource
+      authorize resource
+      resource.save!
+      respond_with root_resource(resource), show_options
+    end
+  end
+
+  def destroy
+    destroy_resource.tap do |resource|
+      authorize resource
+      resource.destroy
+      redirect_to destroy_redirect
+    end
+  end
+
+  private
+
+  def destroy_redirect
+    { action: 'index' }
+  end
+
+  def index_options
+    valid_params_for_action(:index).tap do |options|
+      options[:_keys] = options.keys
+    end
+  end
+
+  def show_options
+    valid_params_for_action(:show).tap do |options|
+      options[:_keys] = options.keys
+      if self.class.resource_representer
+        options[:represent_with] = self.class.resource_representer
+      end
+    end
+  end
+
+  def valid_params_for_action(action)
+    (params.slice(*self.class.valid_params_list(action)) || {}).tap do |p|
+      p[:zoom] = zoom_param if zoom_param
+    end
+  end
+
+  def zoom_param
+    @zoom_param ||= ->() do
```

Comment:
I know I wrote this, but I wonder if it should just be regular `lambda` syntax for a multi-line lambda - do we have a style preference for that condition?